airbyte-integrations/connectors-performance/destination-harness/src/main/java/io/airbyte/integrations/destination_performance/PerformanceTest.java

```diff
@@ -0,0 +1,248 @@
+    // populateStreamToAllFields(this.catalog, streamToAllFields);
+    // final String streamName0 = sourceConfig.getCatalog().getStreams().get(0).getStream().getName();
+    // final String streamNamespace0 =
+    // sourceConfig.getCatalog().getStreams().get(0).getStream().getNamespace();
+    // final String streamName1 = sourceConfig.getCatalog().getStreams().get(1).getStream().getName();
+    // final String streamNamespace1 =
+    // sourceConfig.getCatalog().getStreams().get(1).getStream().getNamespace();
+    // final String streamName2 = sourceConfig.getCatalog().getStreams().get(2).getStream().getName();
+    // final String streamNamespace2 =
+    // sourceConfig.getCatalog().getStreams().get(2).getStream().getNamespace();
+
+    // final var recordSchemaValidator = new RecordSchemaValidator(
+    // Map.of(
+    // new AirbyteStreamNameNamespacePair(streamName0, streamNamespace0),
+    // sourceConfig.getCatalog().getStreams().get(0).getStream().getJsonSchema(),
+    // new AirbyteStreamNameNamespacePair(streamName1, streamNamespace1),
+    // sourceConfig.getCatalog().getStreams().get(1).getStream().getJsonSchema(),
+    // new AirbyteStreamNameNamespacePair(streamName2, streamNamespace2),
+    // sourceConfig.getCatalog().getStreams().get(2).getStream().getJsonSchema()),
+    // true);
+
+    log.info("Destination starting");
+    source.start(sourceConfig, Path.of(jobRoot));
+    var totalBytes = 0.0;
+    var counter = 0L;
+    final var start = System.currentTimeMillis();
+    log.info("Starting Test");
+    while (!source.isFinished()) {
+      final Optional<AirbyteMessage> airbyteMessageOptional = source.attemptRead();
+      if (airbyteMessageOptional.isPresent()) {
+        final AirbyteMessage airbyteMessage = airbyteMessageOptional.get();
+
+        if (airbyteMessage.getRecord() != null) {
+          totalBytes += Jsons.getEstimatedByteSize(airbyteMessage.getRecord().getData());
+          counter++;
+
+          // validateSchema(recordSchemaValidator, streamToAllFields, unexpectedFields, validationErrors,
+          // airbyteMessage);
+          // airbyteMessage.getRecord().setStream(airbyteMessage.getRecord().getStream() + "SUFFIX");
+          // ((ObjectNode) airbyteMessage.getRecord().getData()).retain("id", "user_id", "product_id",
+          // "added_to_cart_at", "purchased_at", "name",
+          // "email",
+          // "title", "gender", "height", "language", "blood_type", "created_at", "occupation", "updated_at",
+          // "nationality");
+          // destination.accept(airbyteMessage);
+        }
+
+      }
+
+      if (counter > 0 && counter % 1_000_000 == 0) {
+        log.info("current throughput: {} total MB {}", (totalBytes / 1_000_000.0) / ((System.currentTimeMillis() - start) / 1000.0),
+            totalBytes / 1_000_000.0);
+      }
+    }
+    log.info("Test ended successfully");
+    final var end = System.currentTimeMillis();
+    final var totalMB = totalBytes / 1_000_000.0;
```

Comment:
we should check with the rest of the performance team to double check our internal definition of a MB, this or 2^20?

Reply:
Good point. [speed dashboard(https://airbyte.metabaseapp.com/question/1824-bottleneck-detector-dataset-serialized-throughput) is dividing the num bytes by 1024^2, so better have it the same in harnesses